### PR TITLE
Refactoring of DockerClientConfig to better match with docker CLI...

### PR DESCRIPTION
**Pull request by marcuslinke**
...configuration options. 

This PR addresses issue #331. It renames all the config options/keys of docker-java  to match better with the docker CLI environment options.

### CODE REVIEW

File: src/main/java/com/github/dockerjava/api/command/DockerCmdExecFactory.java [hunk 1]
```diff
@@ -105,6 +107,8 @@
 
     public DisconnectFromNetworkCmd.Exec createDisconnectFromNetworkCmdExec();
 
+    public DockerCmdExecFactory withSSLContext(SSLContext sslContext);
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/DockerCmdExecFactory.java]: what is the idea of adding it here?

Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/api/command/DockerCmdExecFactory.java]: Passing a custom `SSLContext` to the `DockerClientConfig` could logically collide with the DOCKER_TLS_VERIFY/dockerTlsVerify() option in the sense of consistency. Additionally I want to free `DockerClientConfig` from non-default/special configuration options so I moved it to the `DockerCmdExecFactory`. Makes sense?

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/DockerCmdExecFactory.java]: Seems i totally confused with all this classes... Feel free to merge.

**Comment by KostyaSha** [comment]
@marcuslinke imho it makes sense firstly made refactoring of all `hash` methods and `toString()`. Can i get some window between your refactorings?

File: src/main/java/com/github/dockerjava/core/DockerClientConfig.java [hunk 2]
```diff
@@ -279,11 +280,11 @@ public AuthConfig effectiveAuthConfig(String imageName) {
     }
 
     public AuthConfigurations getAuthConfigurations() {
-        String dockerCfgFile = getDockerCfgPath();
-        if (dockerCfgFile != null) {
+        File dockerCfgFile = new File(getDockerConfig() + File.separator + DOCKER_CFG);
+        if (dockerCfgFile.exists()) {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/DockerClientConfig.java]: `&& isFile()` afair more exact check

File: src/main/java/com/github/dockerjava/jaxrs/DockerCmdExecFactoryImpl.java [hunk 3]
```diff
@@ -146,15 +152,20 @@ public void init(DockerClientConfig dockerClientConfig) {
...
         String protocol = null;
 
-        if (dockerClientConfig.getSslConfig() != null) {
+        if (dockerClientConfig.getDockerTlsVerify()) {
             protocol = "https";
+
             try {
-                sslContext = dockerClientConfig.getSslConfig().getSSLContext();
+
+                if(sslContext == null) {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/jaxrs/DockerCmdExecFactoryImpl.java]: `if (`

**Comment by marcuslinke** [comment]
Closed in favor of rebased version: https://github.com/docker-java/docker-java/pull/447

**Pull request closed without merging**